torchprep/torchprep/torchprep/main.py

```diff
@@ -0,0 +1,100 @@
+import typer
+import torch
+from typing import List
+import time
+import os
+from enum import Enum
+from pathlib import Path
+import math
+
+
+app = typer.Typer()
+
+class Precision(Enum):
+    int8 = "int8"
+    float16 = "float16"
+
+class Device(str, Enum):
+    cpu = "cpu"
+    gpu = "gpu"
+
+
+@app.command()
+def hello() -> None:
+    typer.echo(f"Hello torchprep")
+
+@app.command()
+def distill(model_path : Path, device : Device, parameter_scaling : int, layer_scaling : int = None, profile : List[int] = None) -> None:
+    """
+    Create a smaller student model by setting a distillation ratio and teach it how to behave exactly like your existing model
+    """
+    typer.echo(f"Coming soon")
+    typer.echo("See this notebook for more information https://colab.research.google.com/drive/1RzQtprrHx8PokLQsFiQPAKzfn_DiTpDN?usp=sharing")
+
+@app.command()
+def quantize(model_path : Path, precision : Precision , device : Device = Device.cpu, profile : str = typer.Option(default=None, help="Comma seperated input tensor shape")) -> None:
+    # TODO: define model output path
+    """
+    Quantize a saved torch model to a lower precision float format to reduce its size and latency
+    """
+    model = load_model(model_path, device)
+
+    if device == Device.cpu:
+        if precision == "int8":
+            dtype = torch.qint8
+        else:
+            dtype = torch.float16
+    quantized_model = torch.quantization.quantize_dynamic(
+    model, {torch.nn.LSTM, torch.nn.Linear, torch.nn.Conv2d}, dtype=dtype
```

Comment:
Cool! I can definitely see this as being profitable for `LSTM` and `Linear` layers. I'm wondering about `Conv2d`. Since the balance of parameters/activations is usually flipped for Conv compared to the other two (conv has larger activations and smaller parameters), dynamic quantization might incur high overhead in CNN cases. Do you have any insight on this?

Reply:
I see the `profile` stuff below. Is the idea that this is a human-in-the-loop tool and that the user would be checking that the perf is actually better?